src/topology/algebra/field.lean

```diff
@@ -1,18 +1,118 @@
 /-
-Copyright (c) 2021 Scott Morrison. All rights reserved.
+Copyright (c) 2021 Patrick Massot. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
-Authors: Scott Morrison
+Authors: Patrick Massot, Scott Morrison
 -/
 import topology.algebra.ring
 import topology.algebra.group_with_zero
 
 /-!
-A topological field is usually described via
-`{𝕜 : Type*} [field 𝕜] [topological_space 𝕜] [topological_ring 𝕜] [has_continuous_inv 𝕜]`.
+# Topological fields
+
+A topological division ring is a topological ring whose inversion function is continuous at every
+non-zero element.
 
-The only construction in this file doesn't need to assume `[has_continuous_inv 𝕜]`.
 -/
 
+
+namespace topological_ring
+open topological_space function
+variables (R : Type*) [ring R]
+
+variables  [topological_space R]
+
+/-- The induced topology on units of a topological ring.
+This is not a global instance since other topologies could be relevant. Instead there is a class
+`induced_units` asserting that something equivalent to this construction holds. -/
+def topological_space_units : topological_space (units R) := induced (coe : units R → R) ‹_›
```

Comment:
```suggestion
`induced_units` asserting that something equivalent to this construction holds.

See note [reducible non-instances]. -/
@[reducible]
def topological_space_units : topological_space (units R) := induced (coe : units R → R) ‹_›
```
```

Reply:
I don't see how that is relevant so I'm tempted not to do that until some use appear.

Reply:
Perhaps @fpvandoorn can comment on whether this matters, since they wrote the library note.

Reply:
It is unlikely to come up, so I'm fine with not making this change.

A hypothetical situation this can come up:
Maybe under some conditions on `R`, we have `uniform_space (units R)`, and so we have a similar `uniform_space_units` definition.
Under some other conditions, we might also have `t2_space (units R)`
If we now want to apply a lemma about uniform T2 spaces, type-class inference needs to instantiate this instance (`@t2_space (units R) topological_space_units`) with `@t2_space (units R) (@uniform_space.to_topological_space _ uniform_space_units)`. To do this, these definitions have to be reducible
(the classes `uniform_space` and `t2_space` could be replaced by other classes, I don't know if these make mathematical sense)

Reply:
Something that might be relevant here - `note [reducible non-instances]` is primarily about definitions used as `instance foo := some_def_to_make_the_instance`, whereas what Patrick is doing is `local attribute [instance] some_def_to_make_the_instance`.

So on the one hand, within Patrick's sections the instance is already reducible so this isn't needed - but if someone ends up with a term containing this instance in a scope where it's not locally an instance, then it won't be reducible.

Perhaps we should extend the library note to explicitly include or exclude "instances which are too dangerous to enable globally" (in a separate PR of course!)

Reply:
Let's keep it like this then, and we'll see later.